File: Easy/Python/1379. Find a Corresponding Node of a Binary Tree in a Clone of That Tree.py
class Solution(object):
    def getTargetCopy(self, original, cloned, target):
        # Base case
        if original is None:
            return None
        
        # Current node is cloned
        if original is target:
            return cloned
        
        # See if either left tree or right tree has a target
        return self.getTargetCopy(original.left, cloned.left, target) or self.getTargetCopy(original.right, cloned.right, target)

File: Easy/JavaScript/1379. Find a Corresponding Node of a Binary Tree in a Clone of That Tree.js
var getTargetCopy = function(original, cloned, target) {
    // Base case
    if(original == null)
        return null;
    
    // Current node is cloned
    if(original == target)
        return cloned;
    
    // See if left tree or right tree has a target
    return getTargetCopy(original.left, cloned.left, target) ||
           getTargetCopy(original.right, cloned.right, target);
};
